scripts/pylintrc

```diff
@@ -59,6 +59,8 @@ score=no
 disable=all
 # Identifiers are in the same order as in 'pylint3 --list-msgs'
 enable=
+    fatal,
+#    E, # maybe some day
```

Comment:
Did you check that this doesn't accidentally disable all the tests that follow (`empy-docstring`, etc.)?

`pylint` seems to silently ignore stuff it doesn't understand here sometimes.

Reply:
Should also put `fatal` where it appears in the output of `pylint --list-msgs` (see the comment above the `enable=` line).

What's E btw?

Reply:
IIRC, I tried to enable `fatal`, but it causes a bunch of issues to due modules not being found, etc.

Reply:
Then again, might be nice to do a test run with it, to try to fix the cases that can be fixed. Would be nice if we could get all of them fixed, so that everything can be linted.

Reply:
> Did you check that this doesn't accidentally disable all the tests that follow (empy-docstring, etc.)?

Yes, this syntax is valid.

> What's E btw? 

errors, see the bottom of `man pylint`. Unlike `fatal`, only the initial works with pylint 2.1.1. No idea why.

> IIRC, I tried to enable fatal, but it causes a bunch of issues to due modules not being found, etc.

These are errors, you've been testing `E` by mistake. They are the reason why I commented it out.

Reply:
> errors, see the bottom of `man pylint`. Unlike `fatal`, only the initial works with pylint 2.1.1. No idea why.

Oh... didn't know that `--enable` accepted categories and stuff too, but seems it does. Got confused since it's not listed in `--list-msgs`, but it's a different thing then.

> These are errors, you've been testing `E` by mistake. They are the reason why I commented it out.

Hmm... what do you mean by "testing `E` by mistake"? It was never enabled.

Did you notice any stuff that could be fixed with the E category enabled?

Reply:
> Did you notice any stuff that could be fixed with the entire E category enabled?

Not sure what you mean exactly by "could be fixed" but yes, enabling the entire Error category does report extra errors. I only gave Error a quick try and didn't really pay much attention to them out of "yak shaving" fear: the only purpose of this series is to make it easy to run `check_compliance.py` on `ci-tools` itself + fix what made that difficult along the way (which is already some yak shaving depending on how you look at it)

Now if you're interested, looking at all Errors is as easy as `pylint --rcfile=tools/ci-tools/scripts/pylintrc $(git ls-files \*.py)`. The two most common types seem to be E0401 and E1101
```
************* Module device
boards/xtensa/intel_s1000_crb/support/device.py:17:0: E0401: Unable to import 'periphery' (import-error)
************* Module genrest
doc/scripts/genrest.py:110:23: E1101: Instance of 'str' has no 'prompt' member (no-member)
```

Reply:
> Hmm... what do you mean by "testing E by mistake"?

I mean enabling "fatal" doesn't show any extra... fatality for me (except when bending check_compliance.py over backwards, see commit message). You?

Reply:
Sorry, hadn't checked the commit message. I get it now. I had misunderstood the purpose of `fatal`.

Not making excuses, but this is why I always paste all the commit messages into the PR message, because I assume other people have my flaws too. :)

Reply:
I didn't copy/paste all the commit messages partly for laziness and as a small and impulsive act of rebellion against the generally negligent way github deals with commit messages. For instance they can't even be reviewed. At the very least github could pre-can the description with that copy/paste itself.

On the other hand, the only thing I put in the description is: "See commit messages" :-)

Reply:
Yeah, I'll just blame GitHub for making me lazy by not making it easier to view all the commit messages. ;)

Would be nice if they popped up in some window you could scroll through or something.